spec/new_inject_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'new_inject'
+
+describe Array do
+
+  array1 = [1,2,3,5]
+  array2 = [['hello','world'],['atom','good'],['sublime','bad']]
+
+  describe "#new_inject" do
+    it {expect(subject).to respond_to(:new_inject).with(1).argument}
+
+    it 'produce an array of squares of an array of numbers' do
+      result = array1.new_inject([]) do |obj, ele|
+        obj << ele * ele
+      end
+      norm_inj_res = array1.inject([]) do |obj, ele|
+        obj << ele * ele
+      end
+      expect(result).to eq(norm_inj_res)
+    end
+
+    it 'sums an array of integers when passed an argument of 0' do
+      result = array1.new_inject(0) {|obj, ele| obj + ele}
+      norm_inj_res = array1.inject(0) {|obj, ele| obj + ele}
+      expect(result).to eq(norm_inj_res)
+    end
+
+    it 'sums an array of integers when passed no argument' do
+      result = array1.new_inject() {|obj, ele| obj + ele}
+      norm_inj_res = array1.inject(0) {|obj, ele| obj + ele}
+      expect(result).to eq(norm_inj_res)
+    end
+
+    it 'sums an array of integers when passed no argument and does not change
+      the original array' do
+      array1_original = array1.dup
+      result = array1.new_inject() {|obj, ele| obj + ele}
```

Comment:
Leaving the () for clarity / readability sake, new_inject needs to be able to take no argument in which case it takes the first element of the array as its argument, putting () in the test make this more explicitly readable.